### PR TITLE
KFSPTS-21558 Update KEW-upgraded KFS to run locally

### PR DESCRIPTION
**Pull request by cah292**
This PR updates the KEW-to-KFS version of cu-kfs so that it can run in a local environment.

I am adding new areas on the following Confluence page to document the more notable changes for this effort:

https://confluence.cornell.edu/x/ksfQFw

### CODE REVIEW

**Review by db483 — CHANGES_REQUESTED** [review]
The code changes look good and the application starts locally but I have a few questions and suggestions:

- files in src/test/resources/edu/cornell/kfs/krad/service/impl:
- - Do they need to be updated? have references to rice packages.
- cu-spring-rass-service-test.xml
- - does this need to be updated to use documentProcessingQueue rather than rice.kew.documentProcessingQueue?
- DD files still referencing: xmlns:dd="http://rice.kuali.org/dd". Should this be removed?

Some notes on the confluence updates:
- Module Config and Module Service bean overrides:
- - Did this have to be updated to get the application to start or was it just an improvement?
- for the updated doc types, do they need to be reingested?
- Modules Lock/Unlock Job changes:
- - What was the coreService module?
- - do we have a user story to update the opdoc?
- Notable config property changes:
- - This contains some changes that were just a cleanup and not necessary for the kew upgrade - for example moving the import of the module spring files to the config file. I think it would be best if we could leave such cleanups out in the future as it makes it less clear what changes are part of the upgrade and what changes are just cleanups/improvements.
- - Should we document the new config properties files? Could be done in a separate user story. 
- - Should we create a new PR on the config repo for the new config properties files? Or should that be done when we will work on getting the unit tests to run?

Thank you.

File: src/main/java/org/kuali/kfs/ksb/messaging/threadpool/KSBThreadPoolImpl.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import org.kuali.kfs.core.impl.config.property.Config;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A Thread Pool implementation for the KSB which implements a thread pool backed by a configuration store.
```
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/java/org/kuali/kfs/ksb/messaging/threadpool/KSBThreadPoolImpl.java:34]: KualiCo removed almost all of the KSB code for the KEW-to-KFS upgrade. However, there were a few sections of KSB that KFS still relied on and were kept in place for the upgrade, such as the message queue and a KSB-specific thread pool. This overlayed class is part of the KSB code that had been kept.

File: src/main/config/CU_Checkstyle.xml [hunk 2]
```diff
@@ -69,24 +71,39 @@
...
+
+
+            The following classes are considered illegal within the KFS codebase; they cannot be imported. These
+            classes live in otherwise legal (sub-)packages; however, they generally contain functionality similar to
+            another legal class and it was decided to prefer the other class.
+
+            org.springframework.util.StringUtils     Use commons-lang3's StringUtils.
+        -->
+        <!--
+               For KEW, temporarily allowing the following imports
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/config/CU_Checkstyle.xml:87]: Thank you for explaining this, I am ok with keeping this. Let's ad this to the list of questions to KualiCo.

File: src/main/java/org/kuali/kfs/ksb/messaging/threadpool/KSBThreadPoolImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/org/kuali/kfs/ksb/messaging/threadpool/KSBThreadPoolImpl.java:34]: Thank you for explaining this.

File: src/main/resources/edu/cornell/kfs/sys/cu-spring-sys.xml [hunk 3]
```diff
@@ -325,18 +309,13 @@
 	<bean id="modulesUnlockJob" parent="modulesUnlockJob-parentBean">
 		<property name="steps">
 			<list merge="true">
-				<ref bean="unlockCoreServiceModuleStep"/>
 				<ref bean="unlockSECModuleStep"/>
 				<ref bean="unlockCRModuleStep"/>
 				<ref bean="unlockTAXModuleStep"/>
 			</list>
 		</property>
 	</bean>
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/resources/edu/cornell/kfs/sys/cu-spring-sys.xml:318]: Thank you for explaining this. Keeping a dummy service sounds good. Please create a user story to clean it up when we are ready.

**Comment by cah292** [comment]
Regarding the other points of concern:

* files in src/test/resources/edu/cornell/kfs/krad/service/impl
  * These files are related to unit testing the maintenance XML conversion feature. We'll need these for validating the upcoming changes that will convert the Rice/Cynergy classnames over to the KFS classnames, so we would expect them to have Rice references. One of the user stories in the current sprint will create follow-up stories for implementing the needed changes.
* cu-spring-rass-service-test.xml
  * Good catch, I will update the documentProcessingQueue bean reference accordingly.
* References to http://rice.kuali.org/dd
  * Another good catch, I will remove those references from our Spring bean files.
* Module config/service bean changes:
  * I made these updates for two reasons: To clean up the areas where we were potentially loading certain modules twice, and as a safeguard to prevent any potential problems from such duplication as a result of the KEW-to-KFS upgrade changes from KualiCo. I am not certain whether or not the application launches with the old duplicate module setup in place.
* Doc type updates:
  * The SQL upgrade scripts should convert the KEW tables' various Rice references over to KFS references accordingly. I don't think we'll be required to reingest any existing doc types for the upgrade, unless we discover unusual document workflow problems during testing.
* Modules Lock/Unlock Job:
  * See my earlier comment concerning the CoreService module. I will create a user story to remove the temporary no-op step and to get the OpDoc updated.
* Config changes:
  * Sorry about the confusion from the areas that were meant only as cleanup. If you need me to undo any cleanup, please let me know.
  * I will create a new user story for documenting the notable changes to the config property files for this upgrade.
  * We don't need a PR for the config repo just yet. We will handle that once we're ready to enable PR-triggered unit tests for the upgraded code.

**Comment by cah292** [comment]
I've made some code and documentation changes based on the initial feedback, and have created some follow-up tickets and linked them to the related user story. In addition, I noticed a DB-sequence-related issue during some local testing today, so I've created and linked a follow-up ticket to get that dealt with.

This is ready for review again.

**Pull request merged**